crates/matrix-sdk-indexeddb/src/media_store/transaction.rs

```diff
@@ -242,8 +242,9 @@ impl<'a> IndexeddbMediaStoreTransaction<'a> {
         policy: MediaRetentionPolicy,
     ) -> Result<(), TransactionError> {
         self.put_item_if(media, |indexed| {
-            !indexed.content_size.ignore_policy().is_yes()
-                && !policy.exceeds_max_file_size(indexed.content_size.content_size() as u64)
+            indexed.content_size.ignore_policy().is_yes()
+                || (!indexed.content_size.ignore_policy().is_yes()
+                    && !policy.exceeds_max_file_size(indexed.content_size.content_size() as u64))
```

Comment:
This conceptually does: `A || (!A && B)`. You can simplify it to `A || B`, because the second branch will only be taken if the left hand side is false.

So, comparing to the initial code: only need to remove the leading `!` line 245.

Reply:
fixup: [`c971a2c597bd195fcafb8611163704f5c7fd8791`](https://github.com/matrix-org/matrix-rust-sdk/pull/5749/commits/c971a2c597bd195fcafb8611163704f5c7fd8791)